src/core/usecases/runningService.ts

```diff
@@ -135,6 +140,38 @@ export const thunks = {
 
             dispatch(actions.isUserWatchingChanged({ isUserWatching }));
         },
+    "upgradeService":
+        (serviceId: string): ThunkAction<void> =>
+        async (...args) => {
+            const [dispatch, getState] = args;
+
+            {
+                const state = getState().runningService;
+
+                if (state.isUpdating) {
+                    return;
+                }
+            }
+            const restorableRunningPackageConfigs =
+                runningPackageConfigSelectors.restorableRunnigPackageConfigs(getState());
+            const restorableRunningPackageConfig = restorableRunningPackageConfigs.find(
+                e => e.runningPackageConfig.name === serviceId
+            )?.runningPackageConfig;
+            assert(restorableRunningPackageConfig !== undefined);
+
+            await dispatch(
+                launcherThunk.initialize({
+                    catalogId: restorableRunningPackageConfig.catalogId,
+                    packageName: restorableRunningPackageConfig.packageName,
+                    name: restorableRunningPackageConfig.name,
+                    formFieldsValueDifferentFromDefault:
+                        restorableRunningPackageConfig.formFieldsValueDifferentFromDefault
```

Comment:
We use double quote for object property in this codebase.  
You may disagree but it's important that the coding style be enforced throughout the codebase.